lib/Models/RegionProviderList.js

```diff
@@ -0,0 +1,135 @@
+'use strict';
+
+/*global require*/
+var defined = require('terriajs-cesium/Source/Core/defined');
+var loadJson = require('terriajs-cesium/Source/Core/loadJson');
+var when = require('terriajs-cesium/Source/ThirdParty/when');
+var DeveloperError = require('terriajs-cesium/Source/Core/DeveloperError');
+
+var RegionProvider = require('./RegionProvider');
+
+/**********************************************************************************************
+*  Region mapping support - turn CSVs into choropleths of regions like local government areas
+*  Implements most of http://terria.io/DataProviders/CSV-geo-au
+*  How this works:
+*  1. If there are 'lat' and 'lon' columns, use those instead to generate a point display.
+*  2. Look through all the column names in the CSV file until we find one that matches a defined region descriptor in 
+      region mapping JSON file (not included in TerriaJS)
+*  3. Contact the WFS server defined in that file, fetch IDs for all regions of that type
+*  4. Based on values in the region variable column, generate a linear choropleth mapping
+*  5. Fetch specially prepared WMS tiles which are coloured with a unique colour per region.
+*  6. Recolor each tile, using the unique colour to determine its ID, then replacing that unique colour with a mapped color.
+*
+* Voilà - very fast client-side choroplething with no user data sent to servers, and no vector boundaries sent to client. */
+
+
+/**
+  RegionProviderList encapsulates the regionMapping.json file and provides support for choosing the best region
+  provider for a given dataset.
+*/
+
+var RegionProviderList = function(terria) {
+    /** 
+     * URL pointing to a JSON file where the region provider definitions will be loaded from. 
+     * @type {String}
+     */
+    this.terria = terria;
```

Comment:
This isn't used for anything other than getting the `regionMappingDefinitionsUrl`.  `RegionProvider` doesn't really use `Terria` or any other models, either.

So I think it would make sense to move both `RegionProvider` and `RegionProviderList` to `Map`.

Then, change `RegionProviderList` to take the JSON region definition (like `initFromObject`), remove the `_initialized` logic entirely, and add a static `fromRegionMappingDefinitionsUrl` function that returns a promise for a `RegionProviderList`.  `fromRegionMappingDefinitionsUrl` can also cache the loaded instance so that multiple `CsvCatalogItem`s only need to load it once.